src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java

```diff
@@ -166,13 +178,39 @@ public Directory deleteFolder(Account account, long dirId) throws MopsException
             throw new DeleteAccessPermissionException(errorMessage);
         }
 
-        Directory parentDirectory = getDirectory(directory.getParentId());
 
+
+        Directory parentDirectory = null;
+        if (directory.getParentId() != null) {
+            parentDirectory = getDirectory(directory.getParentId());
+        }
         deleteDirectory(directory);
 
+        if (directory.getParentId() == null || isFirstLevel(directory)) {
```

Comment:
```suggestion
        if (parentDirectory == null || parentDirectory.getPermissionsId() != directory.getPermissionsId()) {
```
it is enough to check if our directory has another permissions object than the parent permissions object, as all child directories are gone and siblings won't have the same permissions